app/src/main/java/com/eveningoutpost/dexdrip/Services/G5CollectionService.java

```diff
@@ -563,8 +553,8 @@ private void processNewTransmitterData(TransmitterData transmitterData, long tim
         Sensor.updateBatteryLevel(sensor, transmitterData.sensor_battery_level);
         android.util.Log.i("timestamp create", Long.toString(transmitterData.timestamp));
 
-        BgReading.create(transmitterData.raw_data, transmitterData.filtered_data, this, transmitterData.timestamp);
-        transmitterData.save();
+        BgReading.create(transmitterData.raw_data, filtered_data, this, transmitterData.timestamp);
+
 
     }
 
```

Comment:
The last read, time since logic should probably stay put. When the G5 wakes up, the app may acquire raw data multiple times before it goes back to sleep. I put those checks in place because BgReading create was inserting multiple duplicates.